tests/test_outcomes_api.py

```diff
@@ -136,7 +136,12 @@ def test_happy_path_querying(self):
                     "selected_columns": [],
                     "to_date": to_date,
                     "organization": 1,
-                    "conditions": [["outcome", "=", 0], ["project_id", "IN", [1, 2]]],
+                    "conditions": [
+                        ["outcome", "=", 0],
+                        ["project_id", "IN", [1, 2]],
+                        ["timestamp", ">", from_date],
```

Comment:
I added the explicit timestamp conditions to let the SnQL version of this test pass. It did make me think that we might want a validator for mandatory fields in Snuba as well as in the SnQL SDK. That's a separate PR problem though.

Reply:
We should definitively have a validator for everything we enforce today through extensions. Extensions are either mandatory or fill in a default value and that is what prevents people from running a query on all projects.

On the other hand, why is this a problem on outcomes only and not in the other api tests ?

Reply:
It is a problem for every SnQL API test, but apparently all the other tests pass regardless of whether the timestamp is filtered or not. This test happens to be set up so if you query for every outcome you get 4 results instead of 3.

It feels like the validator needs to be at least partly provided by the Entity, since not every query through SnQL will require a timestamp filter and some entities use organization instead of project. What do you think about replacing the `get_extensions` function in the Entity with a `get_condition_columns` or something like that?